src/lib/containers/DirectDownload.tsx

```diff
@@ -1,24 +1,37 @@
 import React, { useEffect, useState } from 'react'
 import { Icon } from './row_renderers/utils'
 import {
+  BatchFileRequest,
   FileEntity,
+  FileHandle,
+  FileHandleAssociateType,
+  FileHandleAssociation,
   implementsExternalFileHandleInterface,
 } from '../utils/synapseTypes'
-import { SynapseClient } from '../utils'
+import {
+  getFiles,
+  getFileResult,
+  getEntity
+} from '../utils/SynapseClient'
+import IconSvg from './IconSvg'
 
 export type DirectFileDownloadProps = {
   token?: string
   associatedObjectId: string
   entityVersionNumber?: string
+  associatedObjectType?: FileHandleAssociateType
+  fileHandleId?: string
+  displayFileName?: boolean
 }
 
 const DirectDownload: React.FunctionComponent<DirectFileDownloadProps> = (props) => {
 
-  const {token, associatedObjectId, entityVersionNumber} = props
+  const {token, associatedObjectId, entityVersionNumber, associatedObjectType, fileHandleId, displayFileName} = props
   const [isExternalFile, setIsExternalFile] = useState<boolean>(false)
   const [hasFileAccess, setHasFileAccess] = useState<boolean>(false)
   const [fileEntity, setFileEntity] = useState<FileEntity>()
   const [externalURL, setExternalURL] = useState<string>()
+  const [fileName, setFileName] = useState<string>("")
   let mounted:boolean = true
 
   useEffect( () => {
```

Comment:
This could be modified to execute the effect if inView changes (and only getFileEntityFileHandle() if mounted and inView are true)

Reply:
Done.